### DISCUSSION TITLE
denied by access list

### DISCUSSION THREAD

**Discussion by ZombieXLX** [KSmanis/docker-gentoo-distcc#2] [Q&A] [answered]
~$ docker pull ksmanis/gentoo-distcc:tcp
~$ docker run -d -p 3632:3632 --name gentoo-distcc-tcp --rm ksmanis/gentoo-distcc:tcp
~$ docker logs gentoo-distcc-tcp
distccd[8] (dcc_check_client) ERROR: connection from client 'X.X.X.X:37930' denied by access list

~# distcc-config --get-hosts 
X.X.X.X:3632

~# distcc gcc -c main.c -o main.o
distcc[15531] (dcc_writex) ERROR: failed to write: Connection reset by peer
distcc[15531] Warning: failed to distribute main.c to X.X.X.X:3632, running locally instead

**Comment by KSmanis** [comment, marked as answer]
The error from distccd indicates that your IP was denied, most probably by the `--allow-private` flag:

> --allow-private
>               Allow private networks (10.0.0.0/8, 192.168.0.0/16, 172.16.0.0/12, and 127.0.0.0/8).

If this is the case, you should consider using the `ssh` image variant.

**Reply by ZombieXLX** [reply]
>using the ssh image variant
Ok, it's work for me! Thanks <3
